Classes/Database/RteImagesDbHook.php

```diff
@@ -199,8 +199,7 @@ public function transform_db($value)
                             if ($extension === 'jpg' || $extension === 'jpeg' || $extension === 'gif' || $extension === 'png') {
                                 $fileName = GeneralUtility::shortMD5($absoluteUrl) . '.' . $pI['extension'];
                                 // We insert this image into the user default upload folder
-                                list($table, $field) = explode(':', $this->elRef);
-                                $folder = $GLOBALS['BE_USER']->getDefaultUploadFolder($this->recPid, $table, $field);
+                                $folder = $GLOBALS['BE_USER']->getDefaultUploadFolder();
```

Comment:
How is this related to 'Fix multiple PHP 8.0 warnings'?

Reply:
Accessing undefined properties issues an warning in PHP 8.0
elRef and recPid are long gone and non existing properties in TYPO3 11, which is the only version your master branch targets.

Reply:
I also agree that this is a dirty „fix“ and it should be solved better instead of just deleting the variables.

Actually, it was a quick way of getting the extension to work today, and I wanted to share my changes.

I will try and fix this properly, but not this evening.

I found even more problems in this class - in fact PhpStorm highlighted the class like a christmas tree, so I‘ll probably have to take a broader look.

Reply:
thanks for your work.

Reply:
I've looked into this, and I don't think there is a way to get the record pid, table or field at this point.
It used to be available in RteHtmlParser until https://review.typo3.org/c/Packages/TYPO3.CMS/+/63070

We might be able to get the PID and table name somewhere from the request, but the current field name is known at https://github.com/TYPO3/typo3/blob/22f64585c06af04d0cf95d704d79e7afe9a66d4e/typo3/sysext/core/Classes/DataHandling/DataHandler.php#L1274 and beyond. It used to get passed into RteHtmlParser->init() (until change 63070, see above) and is not available globally.

As the core calls getDefaultUploadFolder() without parameters on various other places too, I don't think it's too much of a hassle to go this way too. In fact, it doesn't change the existing behaviour and just suppresses the PHP warnings.